### DISCUSSION TITLE
How to apply a global style to a parent widget such that all its nodes will inherit that style

### DISCUSSION THREAD

**Discussion by proneon267** [beeware/toga#1858] [Q&A] [answered]
Hello
Suppose, I have a `toga.Box()` called `main_box` and I applied a style to it, like `main_box = toga.Box(style=Pack(font_size=32))`. How do I ensure that the style will be inherited by all its nodes?
For example:
```
main_box = toga.Box(style=Pack(font_size=32))
box1 = toga.Box()
box1_label = toga.Label(text="Hello from box1_label")
box1.add(box1_label)
main_box.add(box1)
```

Here, `box1_label` doesn't have the `font_size` style applied to it.

As a workaround I can apply a style globally with:
```
for widget in app.widgets:
    widget.style.update(**Pack(font_size=32))
```

But it uses `app.widgets`, meaning doing this before setting `app.main_window.content` to `main_box`, won't work.

**Comment by freakboy3742** [comment, marked as answer]
There isn't a "cascading" aspect to the implementation of Toga's Pack style engine. The closest you'll get is to either apply the same style to every widget (in the style you've done), or to create a single style definition, and apply that definition to every object:
```
style = Pack(font_size=32)
main_box = toga.Box(style=style)
label = toga.Label(text="Hello", style=style)
```

A full "Cascading" stylesheet isn't something we're likely to add to Pack, as we're deliberately trying to keep Pack simple. However, it *will* be required by Colosseum, which aims to be a full CSS3 implementation. Colosseum isn't sufficiently mature to be used in practice; but it's part of the "long term" picture for BeeWare.
